src/arviz_stats/psens.py

```diff
@@ -0,0 +1,213 @@
+            "latent-obs_var", sample_dims=("__sample__",)
+        ).sum("latent-obs_var")
+
+    component_draws = component_draws.unstack()
+
+    # calculate lower and upper alpha values
+    lower_alpha = 1 / (1 + delta)
+    upper_alpha = 1 + delta
+
+    # calculate importance sampling weights for lower and upper alpha power-scaling
+    lower_w = np.exp(component_draws.azstats.power_scale_lw(alpha=lower_alpha)).values.flatten()
+    lower_w = lower_w / np.sum(lower_w)
+
+    upper_w = np.exp(component_draws.azstats.power_scale_lw(alpha=upper_alpha)).values.flatten()
+    upper_w = upper_w / np.sum(upper_w)
+
+    return dt.azstats.power_scale_sens(
+        lower_w=lower_w,
+        upper_w=upper_w,
+        delta=delta,
+    )
+
+
+def psens_summary(data, threshold=0.05, round_to=3):
+    """
+    Compute the prior/likelihood sensitivity based on power-scaling perturbations.
+
+    Parameters
+    ----------
+    data : DataTree
+    threshold : float, optional
+        Threshold value to determine the sensitivity diagnosis. Default is 0.05.
+    round_to : int, optional
+        Number of decimal places to round the sensitivity values. Default is 3.
+
+    Returns
+    -------
+    psens_df : DataFrame
+        DataFrame containing the prior and likelihood sensitivity values for each variable
+        in the data. And a diagnosis column with the following values:
+        - "prior-data conflict" if both prior and likelihood sensitivity are above threshold
+        - "strong prior / weak likelihood" if the prior sensitivity is above threshold
+        and the likelihood sensitivity is below the threshold
+        - "-" otherwise
+    """
+    pssdp = psens(data, group="log_prior")["posterior"].to_dataset()
+    pssdl = psens(data, group="log_likelihood")["posterior"].to_dataset()
+
+    joined = xr.concat([pssdp, pssdl], dim="component").assign_coords(
+        component=["prior", "likelihood"]
+    )
+
+    n_vars = np.sum([joined[var].size // 2 for var in joined.data_vars])
+
+    psens_df = pd.DataFrame(
+        (np.full((cast(int, n_vars), 2), np.nan)), columns=["prior", "likelihood"]
+    )
+
+    indices = []
+    for i, (var_name, sel, isel, values) in enumerate(
+        xarray_var_iter(joined, skip_dims={"component"})
+    ):
+        psens_df.iloc[i] = values
+        indices.append(labeller.make_label_flat(var_name, sel, isel))
+    psens_df.index = indices
+
+    def _diagnose(row):
+        if row["prior"] >= threshold and row["likelihood"] >= threshold:
+            return "prior-data conflict"
+        if row["prior"] > threshold > row["likelihood"]:
+            return "strong prior / weak likelihood"
+
+        return "-"
+
+    psens_df["diagnosis"] = psens_df.apply(_diagnose, axis=1)
+
+    return psens_df.round(round_to)
+
+
+# get_log_likelihood and get_log_prior functions should be somewhere else
+def get_log_likelihood(idata, var_name=None, single_var=True):
```

Comment:
we have a get log likelihood in https://github.com/arviz-devs/arviz-stats/blob/main/src/arviz_stats/utils.py#L41 already. `utils` is a bad name though